### PR TITLE
Use environment variables to control deployment of API

### PR DESCRIPTION
**Pull request by codingWithRach**
## What does this change?

Uses the following environment variables for the deployment of the API:
- NEWSLETTERS_UI_SERVE - checks for variable = true, or undefined and not in production
- NEWSLETTERS_API_READ_WRITE - checks for variable = true, or undefined and not in production
- NEWSLETTERS_API_READ - checks for variable = true, or undefined and not in production, or READ_WRITE variable = true, or undefined and not in production

Proxy now does not change path, which we specify directly in routes

## How to test

Run `npm install`
For each environment variable, change the value on the command line and invoke `npm run dev` e.g.
`NEWSLETTERS_UI_SERVE=false npm run dev`
Check that there is no access to the relevant options with the UI

## Have we considered potential risks?

Note that there is a problem when selecting the `api test page` option, but this problem already exists in the main branch and therefore will be fixed separately

## Images

` NEWSLETTERS_API_READ_WRITE=false npm run dev`
The wizard  is not displayed:

Screenshot 2023-02-28 at 15 22 16

` NEWSLETTERS_API_READ=false NEWSLETTERS_API_READ_WRITE=false npm run dev`
The list of current newsletters is not displayed:

Screenshot 2023-02-28 at 15 24 14

### CODE REVIEW

File: apps/newsletters-ui/vite.config.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ export default defineConfig({
 				/** @TODO - Read target from env var / param instead of hardcoding */
 				target: 'http://0.0.0.0:3000',
 				secure: false,
-				rewrite: (path) => path.replace(/^\/api/, ''),
+				//rewrite: (path) => path.replace(/^\/api/, ''),
```
[the remark this answers is not in the document]
Comment (idrise) [reply apps/newsletters-ui/vite.config.ts:16]: oops, missed that...will do!

File: apps/newsletters-api/src/register-ui-server.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+		prefix: '/',
+	});
+
+	// Route for serving the index.html file
+	app.get('/index.html', (req, reply) => {
+		const pathToServedFile = path.join(pathToStaticFiles, 'index.html');
+		const stream = fs.createReadStream(path.resolve(pathToServedFile));
+		return reply.type('text/html').send(stream);
+	});
+}
```
[the remark this answers is not in the document]
Comment (idrise) [reply apps/newsletters-api/src/register-ui-server.ts:20]: awwww, you'd be amazed how hard 20 lines of code can be

File: apps/newsletters-api/src/main.ts [hunk 3]
```diff
@@ -1,15 +1,29 @@
...
-registerCurrentStepRoute(app);
-registerDraftsRoutes(app);
+
+if (isServingReadWriteEndpoints) {
+	registerCurrentStepRoute(app);
+}
+if (isServingReadEndpoints) {
+	registerNewsletterRoutes(app);
+	registerDraftsRoutes(app);
+}
```
[the remark this answers is not in the document]
Comment (idrise) [reply apps/newsletters-api/src/main.ts:26]: good spot! I think moving that is very sensible

File: apps/newsletters-api/src/apiDeploymentSettings.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+export const isServingReadWriteEndpoints =
+	(process.env.NEWSLETTERS_API_READ_WRITE === undefined &&
+		process.env.NODE_ENV !== 'production') ||
+	process.env.NEWSLETTERS_API_READ_WRITE === 'true';
+
+export const isServingReadEndpoints =
+	(process.env.NEWSLETTERS_API_READ === undefined &&
+		process.env.NODE_ENV !== 'production') ||
+	process.env.NEWSLETTERS_API_READ === 'true' ||
+	isServingReadWriteEndpoints;
```
[the remark this answers is not in the document]
Comment (idrise) [reply apps/newsletters-api/src/apiDeploymentSettings.ts:25]: no need for local env files

[the remark this answers is not in the document]
Comment (idrise) [reply apps/newsletters-api/src/apiDeploymentSettings.ts:25]: it is a grim looking block, I think i'll refactor that a little

File: apps/newsletters-api/src/app/routes/health.ts [hunk 5]
```diff
@@ -2,10 +2,8 @@ import type { FastifyInstance } from 'fastify';
 
 export function registerHealthRoute(app: FastifyInstance) {
 	/** Health check endpoint */
-	app.get('/healthcheck', () => {
+	app.get('/api/healthcheck', (req, res) => {
 		console.log('====> Health check endpoint called');
```
[the remark this answers is not in the document]
Comment (codingWithRach) [reply apps/newsletters-api/src/app/routes/health.ts:6]: @cemms1 which bit shouldn't be here?

**Pull request merged**